rpc/src/server.rs

```diff
@@ -90,14 +92,21 @@ impl RpcServer {
             .with_pipeline_size(4);
 
         // HTTP and WS server.
-        let method_router =
-            post(handle_jsonrpc::<Option<Session>>).get(handle_jsonrpc_ws::<Option<Session>>);
+        let post_router = post(handle_jsonrpc::<Option<Session>>);
+        let get_router = if enable_websocket {
+            get(handle_jsonrpc_ws::<Option<Session>>)
```

Comment:
If `enable_websocket` is set to `true`, will the client side receive the error message `Used HTTP Method is not allowed. POST or OPTIONS is required` when sending a `GET` HTTP request?

Reply:
no, it's only for HTTP endpoint.
but the error message for Websocket still have a difference, for request like this send to port 28114 (Websocket port):

```console
curl -s -X GET http://127.0.0.1:28114  -H 'Content-Type: application/json' -d '{ "id": 42,   "jsonrpc": "2.0", "method": "get_raw_tx_pool", "params": [] }
```

the old:
```console
WebSocket Protocol Error: Unable to parse WebSocket key.%
```

now:
```console
Connection header did not include 'upgrade'%
```

I think it's not necessary to fix it.